### ISSUE TITLE
[p0] Write the tokenize(batch) function to return labels as the train input_ids shifted to the right

### ISSUE THREAD

**Issue by annh3** [annh3/ai_planning_searching#30]
from transformers import T5Tokenizer, T5ForConditionalGeneration

tokenizer = T5Tokenizer.from_pretrained("google-t5/t5-small")
model = T5ForConditionalGeneration.from_pretrained("google-t5/t5-small")

input_ids = tokenizer("The <extra_id_0> walks in <extra_id_1> park", return_tensors="pt").input_ids
labels = tokenizer("<extra_id_0> cute dog <extra_id_1> the <extra_id_2>", return_tensors="pt").input_ids

# the forward function automatically creates the correct decoder_input_ids
loss = model(input_ids=input_ids, labels=labels).loss
loss.item()
3.7837

https://huggingface.co/docs/transformers/model_doc/t5#transformers.T5Model.forward

```
def tokenize(batch):
    data = tokenizer(batch['text'], padding=True, truncation=True)
    # your code here
    labels = ..... data right shifted and with a pad token
    # your code here end
    return data
```

**Comment by annh3** [comment]
OK, alternatively--

Add a new column to the train_dataset called 'decoder_input_ids' using .copy() on 'input_ids', shift to right and append eos token at the end

https://discuss.huggingface.co/t/how-to-add-a-new-column-to-a-dataset/6453/3

**Comment by annh3** [comment]
from datasets import load_dataset

ds = load_dataset("cosmos_qa", split="train")

new_column = ["foo"] * len(ds)
ds = ds.add_column("new_column", new_column)
